src/main/java/org/dependencytrack/resources/v1/NotificationRuleResource.java

```diff
@@ -255,4 +257,92 @@ public Response removeProjectFromRule(
+            final NotificationRule rule = qm.getObjectByUuid(NotificationRule.class, ruleUuid);
+            if (rule == null) {
+                return Response.status(Response.Status.NOT_FOUND).entity("The notification rule could not be found.").build();
+            }
+            if (rule.getScope() != NotificationScope.PORTFOLIO) {
+                return Response.status(Response.Status.NOT_ACCEPTABLE).entity("Team subscriptions are only possible on notification rules with PORTFOLIO scope.").build();
+            }
+            if (!rule.getPublisher().getName().equals(DefaultNotificationPublishers.EMAIL.getPublisherName())) {
+                return Response.status(Response.Status.NOT_ACCEPTABLE).entity("Team subscriptions are only possible on notification rules with EMAIL publisher.").build();
+            }
+            final Team team = qm.getObjectByUuid(Team.class, teamUuid);
+            if (team == null) {
+                return Response.status(Response.Status.NOT_FOUND).entity("The team could not be found.").build();
+            }
+            final List<Team> teams = rule.getTeams();
+            if (teams != null && !teams.contains(team)) {
+                rule.getTeams().add(team);
+                qm.persist(rule);
+                return Response.ok(rule).build();
+            }
+            return Response.status(Response.Status.NOT_MODIFIED).build();
+        }
+    }
+
+    @DELETE
+    @Path("/{ruleUuid}/team/{teamUuid}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(
+            value = "Removes a team from a notification rule",
+            response = NotificationRule.class
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 304, message = "The rule does not have the specified team assigned"),
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 404, message = "The notification rule or team could not be found")
+    })
+    @PermissionRequired(Permissions.Constants.SYSTEM_CONFIGURATION)
+    public Response removeTeamFromRule(
+            @ApiParam(value = "The UUID of the rule to remove the project from", required = true)
+            @PathParam("ruleUuid") String ruleUuid,
+            @ApiParam(value = "The UUID of the project to remove from the rule", required = true)
+            @PathParam("teamUuid") String teamUuid) {
+        try (QueryManager qm = new QueryManager()) {
+            final NotificationRule rule = qm.getObjectByUuid(NotificationRule.class, ruleUuid);
+            if (rule == null) {
+                return Response.status(Response.Status.NOT_FOUND).entity("The notification rule could not be found.").build();
+            }
+            if (rule.getScope() != NotificationScope.PORTFOLIO) {
+                return Response.status(Response.Status.NOT_ACCEPTABLE).entity("Team subscriptions are only possible on notification rules with PORTFOLIO scope.").build();
+            }
```

Comment:
I used `addProjectToRule` and `deleteProjectFromRule` as an orientation for `addTeamToRule` and `deleteTeamFromRule` and just kept this restriction because our use case is to subscribe specific project members to their projects and inform them about project related notifications , so the `SYSTEM-Scope` was not necessary for us.

Reply:
Yeah, I reckon it makes sense to not accept project associations because system events are never connected to a project.

As you wrote, there are probably cases where folks would want to use this feature for system events, too. I know of a few who want to be notified when NVD mirroring failed for example.